src/accel/SetupOptions.cc

```diff
@@ -6,12 +6,181 @@
 //---------------------------------------------------------------------------//
 #include "SetupOptions.hh"
 
+#include "corecel/io/Logger.hh"
+#include "corecel/math/ArrayUtils.hh"
 #include "geocel/GeantGeoUtils.hh"
+#include "celeritas/field/RZMapFieldInput.hh"
+#include "celeritas/field/UniformFieldData.hh"
+#include "celeritas/inp/FrameworkInput.hh"
+#include "celeritas/inp/Problem.hh"
 
+#include "AlongStepFactory.hh"
 #include "ExceptionConverter.hh"
 
 namespace celeritas
 {
+namespace
+{
+//---------------------------------------------------------------------------//
+
+auto to_input(SDSetupOptions::StepPoint const& sp)
+{
+    inp::GeantSDStepPointAttributes result;
+    result.global_time = sp.global_time;
+    result.position = sp.position;
+    result.direction = sp.direction;
+    result.kinetic_energy = sp.kinetic_energy;
+    return result;
+}
+
+auto to_input(SDSetupOptions const& sd)
+{
+    inp::GeantSensitiveDetector result;
+
+    result.ignore_zero_deposition = sd.ignore_zero_deposition;
+    result.energy_deposition = sd.energy_deposition;
+    result.locate_touchable = sd.locate_touchable;
+    result.track = sd.track;
+    result.pre = to_input(sd.pre);
+    result.post = to_input(sd.post);
+    result.force_volumes = std::set<G4LogicalVolume const*>(
+        sd.force_volumes.begin(), sd.force_volumes.end());
+    result.skip_volumes = std::set<G4LogicalVolume const*>(
+        sd.skip_volumes.begin(), sd.skip_volumes.end());
+
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct system attributes from setup options.
+ */
+inp::System load_system(SetupOptions const& so)
+{
+    inp::System s;
+    if (celeritas::Device::num_devices())
+    {
+        inp::Device d;
+        d.stack_size = so.cuda_stack_size;
+        d.heap_size = so.cuda_heap_size;
+
+        s.device = d;
+    }
+    return s;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * FrameworkInput adapter function.
+ */
+struct ProblemSetup
+{
+    SetupOptions const& so;
+
+    void operator()(inp::Problem&) const;
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Set a Celeritas problem input definition.
+ */
+void ProblemSetup::operator()(inp::Problem& p) const
+{
+    if (!so.geometry_file.empty())
+    {
+        p.model.geometry = so.geometry_file;
+    }
+    p.diagnostics.output_file = so.output_file;
+
+    p.tuning.num_streams = so.get_num_streams();
+
+    {
+        inp::StateCapacity c;
+        c.tracks = so.max_num_tracks * p.tuning.num_streams;
+        c.initializers = so.initializer_capacity * p.tuning.num_streams;
```

Comment:
Why are we multiplying by `num_streams` here?

Reply:
The previous number of initializers was per stream, and the new options are *per process*. In the draft code for building the core params, we then have:
```c++
oc_inp.initializer_capacity = ceil_div(optical_capacity.initializers, num_streams);
```

Reply:
I see, so for the time being we'll keep the `RunnerInput` capacities as per-process and `RunInput`/`SetupProcess` capacities as per-stream?

Reply:
Yep exactly. And see #1591.